hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -619,6 +633,13 @@ private void writeToBuffer(HoodieRecord<T> record) {
       }
     } else {
       long position = shouldWriteRecordPositions ? record.getCurrentPosition() : -1L;
+      if (baseFileInstantTimeForDeletePositions != null) {
+        if (!baseFileInstantTimeForDeletePositions.equals(record.getCurrentBaseFileInstantTime())) {
```

Comment:
is there a possibility of diff records having diff values for `record.getCurrentBaseFileInstantTime()` 

if not, we could simplify this right. just fetch the value from 1st record and use it as baseInstantTime for log block headers.

Reply:
All the records should contain the same base file instant time.  This is just the extra validation to make sure this assumption holds.  If we feel this is redundant, I'll remove it.

Reply:
Removed.